mdLinkChecker/mdLinkChecker.ts

```diff
@@ -0,0 +1,200 @@
+/**
+ * validate relative urls and external links in markdown files
+ *
+ */
+import validator = require("validator");
+import brokenLink = require("broken-link");
+import path = require("path");
+import fs = require("fs");
+import readline = require("readline");
+
+const exec = require("child_process").exec;
+const args = require("yargs").argv;
+
+//Interface for links
+interface Link {
+  address: string;
+  lineNumber: number;
+}
+
+interface FileReport {
+  all: string[];
+  errors: string[];
+}
+
+function executeCommand(command: string): Promise<string> {
+  return new Promise((resolve, reject) => {
+    exec(command, (error, stdout, stderr) => {
+      if (error) {
+        reject(error);
+      }
+      if (stderr) {
+        reject(stderr);
+      }
+      resolve(stdout);
+    });
+  });
+}
+
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+function checkBrokenLinks(url: string, options: any): Promise<boolean> {
+  return new Promise(resolve => {
+    brokenLink(url, options).then(answer => {
+      if (answer) {
+        resolve(true);
+      } else {
+        resolve(false);
+      }
+    });
+  });
+}
+
+// Is this a valid HTTP/S link?
+function isHttpLink(linkToCheck: string): boolean {
+  // Use the validator to avoid writing URL checking logic
+  // eslint-disable-next-line  @typescript-eslint/camelcase
+  return validator.isURL(linkToCheck, { require_protocol: true, protocols: ["http", "https"] }) ? true : false;
+}
+
+function checkLinksCore(file: string, links: Link[]): Promise<FileReport> {
+  return new Promise(resolve => {
+    const fileReport: FileReport = { all: [], errors: [] };
+
+    links.forEach(async (link, index, array) => {
+      let isBroken = false;
+      if (isHttpLink(link.address)) {
+        // Check external links
+        isBroken = await checkBrokenLinks(link.address, { allowRedirects: true, match404Page: /404/ });
+      } else {
+        // Check markdown relative urls
+        try {
+          const currentWorkingDirectory = path.dirname(file);
+          const fullPath = path.resolve(currentWorkingDirectory, link.address).split("#")[0];
+          isBroken = !fs.existsSync(fullPath);
+        } catch (error) {
+          // If there's an error, log the link
+          console.log(`Error: ${link.address} on line ${link.lineNumber} is not an HTTP/s or relative link.`);
+          isBroken = true;
+        }
+      }
+
+      // Print log
+      let message = "";
+      if (isBroken) {
+        message = `Error: [${file}] ${link.address} on line ${link.lineNumber} is unreachable.`;
+        fileReport.errors.push(message);
+        fileReport.all.push(message);
+      } else {
+        message = `Info: [${file}] ${link.address} on line ${link.lineNumber}.`;
+        fileReport.all.push(message);
+      }
+      // console.log(message);
```

Comment:
You could remove this debug information if not used any more

Reply:
Fixed.